packages/app/documents/service.ts

```diff
@@ -725,3 +725,50 @@ function getWorkflowEdgesMatchingSourceAndTarget(
         edge => edge.source === source && edge.target === target
     )
 }
+
+/**
+ * Validates a document against its schema strictly, allowing no additiona properties.
+ */
+export async function strictValidateDocument(
+    documentToValidate: any,
+    modelName: string
+): Promise<ErrorData<Document>> {
+    try {
+        //* Get the model for its schema and title property.
+        const [modelWithWorkflowError, modelWithWorkflow] =
+            await getModelWithWorkflow(modelName, undefined, {
```

Comment:
I'm not sure it matters, but since the workflow isn't used here, would calling `getModel` instead save a query?